system/Validation/Validation.php

```diff
@@ -353,23 +355,33 @@ public function withRequest(RequestInterface $request): ValidationInterface
      *        'rule' => 'message'
      *    ]
      *
-     * @param string      $field
-     * @param string|null $label
-     * @param string      $rules
-     * @param array       $errors
+     * @param string       $field
+     * @param string|null  $label
+     * @param string|array $rules
+     * @param array        $errors
      *
      * @return $this
+     *
+     * @throws TypeError
      */
-    public function setRule(string $field, string $label = null, string $rules, array $errors = [])
+    public function setRule(string $field, string $label = null, $rules, array $errors = [])
     {
-        $this->rules[$field] = [
-            'label' => $label,
-            'rules' => $rules,
+        if (! is_array($rules) && ! is_string($rules)) {
+            throw new TypeError('$rules must be of type string|array');
+        }
+
+        $ruleSet = [
+            $field => [
+                'label' => $label,
+                'rules' => $rules,
+            ],
         ];
 
-        $this->customErrors = array_merge($this->customErrors, [
-            $field => $errors,
-        ]);
+        if ($errors) {
+            $ruleSet[ $field ]['errors'] = $errors;
```

Comment:
```suggestion
            $ruleSet[$field]['errors'] = $errors;
```

Reply:
@paulbalandan is there a CI4 PHPStorm codestyle yml which i can use?

Reply:
I'm not familiar with PHPStorm so I can answer you on that. But the latest version of the ruleset in the `develop` branch will help you fix cs. (Sadly, `4.2` isn't yet updated to latest ruleset)